### PR TITLE
Fixes for the PolynomialGenerator (tutorial)

### PR DESCRIPTION
**Pull request by clavin-xlnx**

### CODE REVIEW

File: src/com/xilinx/rapidwright/gui/GUIModuleInst.java [hunk 1]
```diff
@@ -128,9 +128,10 @@ private void init() {
                 maxCol = (maxCol >= col) ? maxCol : col;
                 maxRow = (maxRow >= row) ? maxRow : row;
                 String tileTypeStr = tile.getTileTypeEnum().toString();
-                if (tileTypeStr.startsWith("BRAM")
-                        || tileTypeStr.startsWith("DSP")) {
+                if (tileTypeStr.startsWith("BRAM")) {
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/gui/GUIModuleInst.java:131]: Consider `Utils.isBRAM()`

File: src/com/xilinx/rapidwright/gui/GUIModuleInst.java [hunk 2]
```diff
@@ -128,9 +128,10 @@ private void init() {
...
                     row = row - 3;
+                } else if (tileTypeStr.startsWith("DSP")) {
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/gui/GUIModuleInst.java:133]: Consider Utils.isDSP()

File: src/com/xilinx/rapidwright/gui/GUIModuleInst.java [hunk 3]
```diff
@@ -167,14 +168,21 @@ private void init() {
             //int tileY = tile.getRow() - minRow;
             int tileX = scene.getDrawnTileX(tile) - minCol;
             int tileY = scene.getDrawnTileY(tile) - minRow;
-            if (tile.getTileTypeEnum().toString().startsWith("BRAM")
-                    || tile.getTileTypeEnum().toString().startsWith("DSP")) {
+            if (tile.getTileTypeEnum().toString().startsWith("BRAM")) {
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/gui/GUIModuleInst.java:171]: Consider `Utils.isBRAM()`

File: src/com/xilinx/rapidwright/gui/GUIModuleInst.java [hunk 4]
```diff
@@ -167,14 +168,21 @@ private void init() {
...
                 hmTileMap[tileY][tileX] = true;
                 hmTileMap[tileY - 1][tileX] = true;
                 hmTileMap[tileY - 2][tileX] = true;
                 hmTileMap[tileY - 3][tileX] = true;
 
-            } else if (tileX >= 0 && tileX < widthInTiles && tileY >= 0
+            } else if (tile.getTileTypeEnum().toString().startsWith("DSP")) {
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/gui/GUIModuleInst.java:177]: Consider `Utils.isDSP()`

File: test/src/com/xilinx/rapidwright/examples/TestPolynomialGenerator.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        Path dcp = dir.resolve("polynomial.dcp");
+
+        PolynomialGenerator.generatePolynomial(polynomial, "test", bitWidth, true, dcp.toString(), null, false);
+
+        Design d = Design.readCheckpoint(dcp);
+
+        for (Cell c : d.getCells()) {
+            if (c.isLocked() || c.isRoutethru())
+                continue;
+            assert (c.isPlaced());
```
Comment (eddieh-xlnx) [inline test/src/com/xilinx/rapidwright/examples/TestPolynomialGenerator.java:53]:
```suggestion
            Assertions.assertTrue(c.isPlaced());
```

File: test/src/com/xilinx/rapidwright/examples/TestPolynomialGenerator.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        DesignTools.updatePinsIsRouted(d);
+        for (Net n : d.getNets()) {
+            for (SitePinInst spi : n.getPins()) {
+                assert (spi.isRouted());
```
Comment (eddieh-xlnx) [inline test/src/com/xilinx/rapidwright/examples/TestPolynomialGenerator.java:59]:
```suggestion
                Assertions.assertTrue(spi.isRouted());
```

File: src/com/xilinx/rapidwright/gui/GUIModuleInst.java (hunk 1 above)
Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/gui/GUIModuleInst.java]: Thanks, good idea.

**Review by eddieh-xlnx — APPROVED** [review]
Thanks!

Don't forget to update the copyright header to 2023 too.

**Pull request merged**